opentasks-provider/src/androidTest/java/org/dmfs/provider/tasks/TaskProviderInstancesTest.java

```diff
@@ -0,0 +1,346 @@
+        RowSnapshot<Instances> instance = new VirtualRowSnapshot<>(new InstanceTable(mAuthority));
+
+        assertThat(new Seq<>(
+                // create a local list
+                new Put<>(taskList, new NameData("list1")),
+                // insert a new task straight into the instances table
+                new Put<>(instance, new Referring<>(Tasks.LIST_ID, taskList, new CharSequenceRowData<>(Tasks.TITLE, "task1"))),
+                // delete the instance
+                new Delete<>(instance)
+        ), resultsIn(mClient,
+                new Assert<>(taskList, new NameData("list1")),
+                // the list does not contain a single task
+                new AssertEmptyTable<>(new TasksTable(mAuthority)),
+                new AssertEmptyTable<>(new InstanceTable(mAuthority))));
+
+    }
+
+
+    /**
+     * Create a single instance and insert an override for exactly the same instance.
+     */
+    @Test(expected = UnsupportedOperationException.class)
+    public void testInsertSingleInstanceTwice() throws RemoteException, OperationApplicationException
+    {
+        RowSnapshot<TaskLists> taskList = new VirtualRowSnapshot<>(new LocalTaskListsTable(mAuthority));
+        RowSnapshot<Tasks> task = new VirtualRowSnapshot<>(new TasksTable(mAuthority));
+        RowSnapshot<Instances> instance = new VirtualRowSnapshot<>(new InstanceTable(mAuthority));
+
+        DateTime dateTime = DateTime.parse("20180110T224500Z");
+        String dtstart = Long.toString(dateTime.getTimestamp());
+
+        new BaseTransaction().with(new Seq<>(
+                // create a local list
+                new Put<>(taskList, new NameData("list1")),
+                // insert a new task into the tasks table (we insert a task to get a RowReference to the new row)
+                new Put<>(task,
+                        new Composite<>(
+                                new Referring<>(Tasks.LIST_ID, taskList),
+                                new TimeData(dateTime),
+                                new TitleData("task1"))),
+                new Put<>(instance,
+                        new Composite<>(
+                                new Referring<>(Tasks.LIST_ID, taskList),
+                                new CharSequenceRowData<>(Tasks.DTSTART, "1234"),
+                                // insert an instance which would override the original instance, which already exists
+                                new Referring<>(Tasks.ORIGINAL_INSTANCE_ID, task),
+                                new CharSequenceRowData<>(Tasks.ORIGINAL_INSTANCE_TIME, dtstart),
+                                new CharSequenceRowData<>(Tasks.TZ, "UTC"),
+                                new CharSequenceRowData<>(Tasks.TITLE, "task1")))
+
+        )).commit(mClient);
+    }
+
+
+    /**
+     * Create a single instance and insert an override for a new instance, turning the event into a recurring event.
```

Comment:
What is the use-case for this - turning a single task into a recurrent one by inserting an override?
Wouldn't we always create recurring tasks by creating a master first? Or turning an single task into a master by updating it with rules.
And only then could we create an override by selecting one of the recurring instances and updating it, as I imagine.

I am talking about usage from user point of view, in a calendar app or in this task app.

Reply:
This is not about `turning a single task into a recurrent one` it's about allowing to add instances to any task, (already) recurring or not. From the provider client's point of view the instances table is one big space of instances. It doesn't really matter which one is of a recurring task and which one isn't. And it should allow the user to add an instance to any task (which is not an override already). I'm currently considering to add another column to the instances table which contains the master task ID, so you always have direct access. This would make it easier to handle these cases. My goal is to allow the UI to handle recurring and non-recurring tasks mostly the same way (i.e. "non-recurring" as a special case of "recurring").

As for the use case, think about opening a task and having an option to repeat this task once again, i.e. sort of a "manual recurrence".